flocker/acceptance/test_environment_variables.py

```diff
@@ -247,12 +245,15 @@ def get_mysql_node_2(ignored):
             Move MySQL to ``node_2`` and return a ``Deferred`` which fires
             with a connection to the previously created database on ``node_2``.
             """
+            # Listen on different port so it's clear we're connecting to
```

Comment:
`assert_expected_deployment` just checks what the control service *thinks* the current state is. That may not be the same as actual state, which is why we want to test things directly.